src/axom/inlet/SphinxDocWriter.cpp

```diff
@@ -0,0 +1,95 @@
+#include "axom/inlet/SphinxDocWriter.hpp"
+#include <iostream>
+#include <assert.h>
+#include "axom/slic.hpp"
+
+namespace axom
+{
+namespace inlet
+{
+ 
+SphinxDocWriter::SphinxDocWriter(const std::string& fileName, axom::sidre::Group* root) {
+  SLIC_ASSERT_MSG(root != nullptr, "Sidre Group is null pointer");
+  m_sidreRootGroup = root;
+  m_fileName = fileName;
+}
+
+void SphinxDocWriter::writeDocuments(axom::sidre::Group* sidreGroup) {
+  if (sidreGroup->getName() == "") {
+    writeTitle("Untitled");
+  }  else {
+    writeTitle(sidreGroup->getName());
+  }
+  m_rstTable = {{"Field Name", "Description", "Default Value", "Range", "Required"}};
+  writeDocumentsHelper(m_sidreRootGroup);
+  writeTable("Fields");
+  m_outFile.open(m_fileName);
+  m_outFile << m_oss.str();
+  m_outFile.close();
+}
+
+void SphinxDocWriter::writeDocumentsHelper(axom::sidre::Group* sidreGroup) {
+  SLIC_ASSERT_MSG(sidreGroup, "Root was nullptr");
+  axom::sidre::IndexType i = sidreGroup->getFirstValidGroupIndex();
+  if (sidreGroup != m_sidreRootGroup && i == axom::sidre::InvalidIndex) { 
+    // means that it is a field so attributes are stored in views
+    std::vector<std::string> fieldAttributes(5, "Not specified");
+    fieldAttributes.resize(5);
+    fieldAttributes[0] = sidreGroup->getName();
+    if (sidreGroup->hasView("description")) {
+      fieldAttributes[1] = std::string(sidreGroup->getView(sidreGroup->getViewIndex("description"))->getString());
+    } 
+    // fieldAttributes[2] = std::to_string(sidreGroup->getView(sidreGroup->getViewIndex("default values"))->getScalar());

```

Comment:
Please remove commented out code if it is truly not needed or leave a comment of why its commented out.